libensemble/tests/regression_tests/test_persistent_aposmm_exception.py

```diff
@@ -0,0 +1,86 @@
+# """
+# Runs libEnsemble on the 6-hump camel problem. Documented here:
+#    https://www.sfu.ca/~ssurjano/camel6.html
+#
+# Execute via one of the following commands (e.g. 3 workers):
+#    mpiexec -np 4 python3 test_6-hump_camel_persistent_uniform_sampling.py
+#    python3 test_6-hump_camel_persistent_uniform_sampling.py --nworkers 3 --comms local
+#    python3 test_6-hump_camel_persistent_uniform_sampling.py --nworkers 3 --comms tcp
+#
+# The number of concurrent evaluations of the objective function will be 4-1=3.
+# """
+
+# Do not change these lines - they are parsed by run-tests.sh
+# TESTSUITE_COMMS: local mpi
+# TESTSUITE_NPROCS: 4
+
+import sys
+import numpy as np
+
+# Import libEnsemble items for this test
+from libensemble.libE import libE
+from libensemble.sim_funcs.periodic_func import func_wrapper as sim_f
+from libensemble.gen_funcs.persistent_aposmm import aposmm as gen_f
+from libensemble.alloc_funcs.persistent_aposmm_alloc import persistent_aposmm_alloc as alloc_f
+from libensemble.tools import parse_args, add_unique_random_streams
+
+
+def assertion(passed):
+    """Raise assertion or MPI Abort"""
+    if libE_specs['comms'] == 'mpi':
+        from mpi4py import MPI
+        if passed:
+            print("\n\nMPI will be aborted as planned\n\n", flush=True)
+            MPI.COMM_WORLD.Abort(0)  # Abort with success
+        else:
+            MPI.COMM_WORLD.Abort(1)  # Abort with failure
+    else:
+        assert passed
```

Comment:
What does the 'anyway' mean?

Reply:
I added that, partially to trigger a build and partially as a reminder that the PETSC ERROR is expected.

Reply:
"Even though the PETSC ERROR is displayed, the correct exception was caught from aposmm by libensemble" I guess was my thought process.

Reply:
Maybe say "Expected exception received' or something along those lines.